### DISCUSSION TITLE
Question about the sampling in log domain

### DISCUSSION THREAD

**Discussion by li7dragonball** [optuna/optuna#3853] [Q&A] [answered]
Hello
Everyone,I have been testing some simple function of optuna recently, and found a problem about the sampling in the log domain.
In the home page of optuna, there is a simple optimization example, in which the "objective" part is written as follows:
```python
def objective(trial):
    x = trial.suggest_int('x', -10, 10)
    return (x - 2) ** 2
```

In order to verify the function of IntLogUniformDistribution, I have changed it as follows:

```python
def objective(trial):
    x = trial.suggest_int('x', 1, 10, log=True)
    print(x)
    return (x - 2) ** 2
```
The printed "x" is distributed between 1 and 10 rather than 10^1 and 10^10. The sampling space seems to be unchanged although the "log" is set to true, I am not sure whether it is a bug, or there is somewhere I have missed, I deeply appreciate for any help or replies.

**Comment by nzw0301** [comment, marked as answer]
Have you seen a note in `log` argument on https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.Trial.html#optuna.trial.Trial.suggest_int? It says

> If log is true, at first, the range of suggested values is divided into grid points of width 1. The range of suggested values is then converted to a log domain, from which a value is sampled. The uniformly sampled value is re-converted to the original domain and rounded to the nearest grid point that we just split, and the suggested value is determined. For example, if low = 2 and high = 8, then the range of suggested values is [2, 3, 4, 5, 6, 7, 8] and lower values tend to be more sampled than higher values.

**Reply by li7dragonball** [reply]
Thank you for replying the question, so the final value of parameter is still distributed in the original domain because of the re-converted operation.

**Reply by nzw0301** [reply]
Right.

**Comment by DeLaRiva** [comment]
Is there an option or workaround/adjustment to prefer sampling from higher values?

May be via raising to the power of e or so for the input range?

**Reply by li7dragonball** [reply]
I think raising the input range may be the only way to access the higher value
